From 03948e7b3a692708345e2c83a839b22b1c6b3864 Mon Sep 17 00:00:00 2001
From: Ashutosh Chauhan <hashutosh@apache.org>
Date: Mon, 13 Feb 2017 12:50:09 -0800
Subject: [PATCH 0982/1164] HIVE-15895 : Use HDFS for stats collection temp
 dir on blob storage (Ashutosh Chauhan via Sergio
 Pena)

Signed-off-by: Ashutosh Chauhan <hashutosh@apache.org>
(cherry picked from commit 2433fed55bf3417c02551ccee6bb76b282905a13)

Conflicts:
	ql/src/java/org/apache/hadoop/hive/ql/parse/SemanticAnalyzer.java

Change-Id: I9270d6341285103536d5d98df385a180a77df78d
---
 .../hadoop/hive/ql/parse/SemanticAnalyzer.java     |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ql/src/java/org/apache/hadoop/hive/ql/parse/SemanticAnalyzer.java b/ql/src/java/org/apache/hadoop/hive/ql/parse/SemanticAnalyzer.java
index ccfc2c2..a955630 100644
--- a/ql/src/java/org/apache/hadoop/hive/ql/parse/SemanticAnalyzer.java
+++ b/ql/src/java/org/apache/hadoop/hive/ql/parse/SemanticAnalyzer.java
@@ -9669,7 +9669,7 @@ private void setupStats(TableScanDesc tsDesc, QBParseInfo qbp, Table tab, String
       tsDesc.setGatherStats(false);
     } else {
       if (HiveConf.getVar(conf, HIVESTATSDBCLASS).equalsIgnoreCase(StatDB.fs.name())) {
-        String statsTmpLoc = ctx.getExtTmpPathRelTo(tab.getPath()).toString();
+        String statsTmpLoc = ctx.getTempDirForPath(tab.getPath()).toString();
         LOG.info("Set stats collection dir : " + statsTmpLoc);
         conf.set(StatsSetupConst.STATS_TMP_LOC, statsTmpLoc);
       }
-- 
1.7.9.5

